axe_selenium_python/tests/test_axe.py

```diff
@@ -5,33 +5,44 @@
 from os import path
 
 import pytest
+from selenium import webdriver
 
+from ..axe import Axe
 
-@pytest.mark.nondestructive
-def test_execute(axe):
-    """Run axe against base_url and verify JSON output."""
-    axe.inject()
-    data = axe.execute()
-    assert data is not None, data
+_DEFAULT_TEST_FILE = path.join(path.dirname(__file__), 'test_page.html')
 
 
 @pytest.mark.nondestructive
-def test_report(axe):
-    """Test that report exists."""
+def test_run_axe_sample_page_firefox():
+    """Run axe against sample page and verify JSON output is as expected."""
+    driver = webdriver.Firefox()
+    driver.get('file://' + _DEFAULT_TEST_FILE)
+    print(_DEFAULT_TEST_FILE)
+    axe = Axe(driver)
+
     axe.inject()
-    results = axe.execute()
-    violations = results["violations"]
-    report = axe.report(violations)
-    assert report is not None, report
+    data = axe.execute()
+    driver.close()
```

Comment:
If your test fails before it reaches this line, then the browser instance will remain open. For anything that requires teardown I would suggest using pytest fixtures. In this case, you have two mostly identical tests, so a parameterised fixture would be ideal. Here's how it could look:

```python
import pytest
from selenium import webdriver


@pytest.fixture(params=('Firefox', 'Chrome'))
def driver(request):
    driver = getattr(webdriver, request.param)()
    yield driver
    driver.close()


def test_run_axe_sample_page(driver):
    driver.get('file://' + _DEFAULT_TEST_FILE)
    ...
```

You'll only need one test, which will be executed with each browser. If you run the tests with `--verbose` you'll see 'Firefox' and 'Chrome' in the test ids.